lib/index.js

```diff
@@ -224,13 +226,37 @@ Socket.prototype._register = function(chan){
  */
 
 Socket.prototype.bind = function (addr) {
-    return this._protect(nn.Bind(this.binding, addr));
+  var eid = nn.Bind( this.binding, addr );
+  if(eid < 0){
+    return this.emit('error', new Error(nn.Err() +': '+this.type+' bind@' + addr));
+  }
+  this.bound[addr] = eid;
+  return this.bound; //socket's active endpoints
 }
 
 Socket.prototype.connect = function (addr) {
-    return this._protect(nn.Connect(this.binding, addr));
+  var eid = nn.Connect( this.binding, addr )
+  if(eid < 0){
+    return this.emit('error', new Error(nn.Err() +': '+this.type+' connect@' + addr));
+  }
+  this.connected[addr] = eid;
+  return this.connected; //socket's active endpoints
```

Comment:
I'm not found of duplicating functionality from `_protect` for the sake of custom error strings.  Let's ruse `protect`:
```js
Socket.prototype.connect = function (addr) {
  return (this.connected[addr] = this._protect(nn.Connect(this.binding, addr)));
};
```